src/Orleans/Async/Cancellation/GrainCancellationToken.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.CodeGeneration;
+using Orleans.Runtime;
+using Orleans.Serialization;
+
+namespace Orleans.Async
+{
+    /// <summary>
+    /// Grain cancellation token that can be passed thought cross-domain boundaries
+    /// </summary>
+    [Serializable]
```

Comment:
from what I understand, `SerializableAttribute` prompts codegen to generate Orleans serializers for the type. Nevertheless, in this particular case we are the ones already providing the `[SerializerMethod]` and so on... personally I would remove [Serializable] (which can give the wrong impression to users that they could use a binary formatter and persist this in storage), and instead explicitly add it to `SerializationManager.RegisterBuiltInSerializers()`, which is how many other built in objects are registered instead of relying in [Serializable].

re comment: sure, we could call it cross-grain boundaries, as it would be more appropriate, but still is unrelated to the necessity of the Serializable attribute.